spt/features/ihud.cpp

```diff
@@ -7,32 +7,191 @@
 #include "signals.hpp"
 #include "property_getter.hpp"
 #include "..\sptlib-wrapper.hpp"
-#include "Color.h"
 
 InputHud spt_ihud;
 
-ConVar y_spt_ihud("y_spt_ihud", "0", FCVAR_CHEAT, "Draws movement inputs of client.\n0 = Default,\n1 = tas.\n");
-ConVar y_spt_ihud_button_color("y_spt_ihud_button_color",
-                               "0 0 0 233",
-                               FCVAR_CHEAT,
-                               "RGBA button color of input HUD.\n");
-ConVar y_spt_ihud_shadow_color("y_spt_ihud_shadow_color",
-                               "0 0 0 66",
-                               FCVAR_CHEAT,
-                               "RGBA button shadow color of input HUD.\n");
-ConVar y_spt_ihud_font_color("y_spt_ihud_font_color",
-                             "255 255 255 233",
-                             FCVAR_CHEAT,
-                             "RGBA font color of input HUD.\n");
-ConVar y_spt_ihud_shadow_font_color("y_spt_ihud_shadow_font_color",
-                                    "255 255 255 66",
-                                    FCVAR_CHEAT,
-                                    "RGBA button shadow font color of input HUD.\n");
-ConVar y_spt_ihud_grid_size("y_spt_ihud_grid_size", "60", FCVAR_CHEAT, "Grid size of input HUD.\n");
-ConVar y_spt_ihud_grid_padding("y_spt_ihud_grid_padding", "2", FCVAR_CHEAT, "Padding between grids of input HUD.");
-ConVar y_spt_ihud_font("y_spt_ihud_font", "Trebuchet20", FCVAR_CHEAT, "Font of input HUD.");
-ConVar y_spt_ihud_x("y_spt_ihud_x", "2", FCVAR_CHEAT, "X offset of input HUD.\n");
-ConVar y_spt_ihud_y("y_spt_ihud_y", "2", FCVAR_CHEAT, "Y offset of input HUD.\n");
+ConVar y_spt_ihud("y_spt_ihud", "0", 0, "Draws movement inputs of client.\n");
+ConVar y_spt_ihud_grid_size("y_spt_ihud_grid_size", "60", 0, "Grid size of input HUD.\n");
+ConVar y_spt_ihud_grid_padding("y_spt_ihud_grid_padding", "2", 0, "Padding between grids of input HUD.");
+ConVar y_spt_ihud_x("y_spt_ihud_x", "2", 0, "X offset of input HUD.\n");
+ConVar y_spt_ihud_y("y_spt_ihud_y", "2", 0, "Y offset of input HUD.\n");
+
+CON_COMMAND(
+    y_spt_ihud_modify,
+    "y_spt_ihud_modify <element|all> <param> <value> - Modifies parameters in given element.\nParams: enabled, text, font, x, y, width, height, background, highlight, textcolor, texthighlight.\n")
+{
+	if (args.ArgC() != 4)
+	{
+		Msg("Usage: y_spt_ihud_modify <element|all> <param> <value>\n");
+		return;
+	}
+	const char* errorMsg =
+	    "Invalid arguments.\nElements: all, angles, forward, back, moveright, moveleft, use, duck, jump, attack, attack2\nParams: enabled, text, font, x, y, width, height, background, highlight, textcolor, texthighlight.\n";
+
+	const char* element = args.Arg(1);
+	const char* param = args.Arg(2);
+	const char* value = args.Arg(3);
+
+	if (std::strncmp(element, "all", 16) == 0)
```

Comment:
I think strcmp should be safe here (and everywhere else in the code) since at least the second C string is valid
```suggestion
	if (std::strcmp(element, "all") == 0)
```